src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/AttributeComponent.scala

```diff
@@ -190,6 +210,82 @@ trait AttributeComponent {
+        We filter on ta.owner_id != null if we want to see that the row exists in both of the tables
+       */
+
+      def ownerIdTail(ownerIds: Seq[OWNER_ID]) = {
+        concatSqlActions(sql"(", reduceSqlActionsWithDelim(ownerIds.map {
+          case(ownerId: UUID) => sql"$ownerId"
+          case(ownerId: Long) => sql"$ownerId"
+        }), sql")")
+      }
+
+      def deleteFromMasterAction(ownerIds: Seq[OWNER_ID]) =
+        concatSqlActions(sql"""delete ea from #${baseTableRow.tableName} ea
+                left join #${baseTableRow.tableName}_TEMP ta
+                on (ea.name,ea.owner_id,ea.list_index)<=>(ta.name,ta.owner_id,ta.list_index)
+                where ta.owner_id is null and ea.owner_id in """, ownerIdTail(ownerIds)).as[Int]
+      def updateInMasterAction(ownerIds: Seq[OWNER_ID]) =
+        concatSqlActions(sql"""update #${baseTableRow.tableName} ea
+                left join #${baseTableRow.tableName}_TEMP ta
+                on (ea.name,ea.owner_id,ea.list_index)<=>(ta.name,ta.owner_id,ta.list_index)
+                set ea.value_string=ta.value_string, ea.value_number=ta.value_number, ea.value_boolean=ta.value_boolean, ea.value_entity_ref=ta.value_entity_ref
+                where ta.owner_id is not null and ea.owner_id in """, ownerIdTail(ownerIds)).as[Int]
+      def insertIntoMasterAction(ownerIds: Seq[OWNER_ID]) =
+        concatSqlActions(sql"""insert into #${baseTableRow.tableName}(name,value_string,value_number,value_boolean,value_entity_ref,list_index,owner_id)
+                select ta.name,ta.value_string,ta.value_number,ta.value_boolean,ta.value_entity_ref,ta.list_index,ta.owner_id
+                from #${baseTableRow.tableName}_TEMP ta
+                left join #${baseTableRow.tableName} ea
+                on (ea.name,ea.owner_id,ea.list_index)<=>(ta.name,ta.owner_id,ta.list_index)
+                where ea.owner_id is null and ta.owner_id in """, ownerIdTail(ownerIds)).as[Int]
+
+      def createAttributeTempTableAction() = {
+        val prefix = sql"""create temporary table #${baseTableRow.tableName}_TEMP (
+              id bigint(20) unsigned NOT NULL AUTO_INCREMENT primary key,
+              name text NOT NULL,
+              value_string text,
+              value_number double DEFAULT NULL,
+              value_boolean bit(1) DEFAULT NULL,
+              value_entity_ref bigint(20) unsigned DEFAULT NULL,
+              list_index int(11) DEFAULT NULL, """
+
+        val suffix = if(typeOf[OWNER_ID] =:= typeOf[Long]) {
+          sql"""owner_id bigint(20) unsigned NOT NULL)"""
+        } else if(typeOf[OWNER_ID] =:= typeOf[UUID]) {
+          sql"""owner_id binary(16) NOT NULL)"""
+        } else throw new RawlsException("Owner ID was of an unexpected type")
+
+        concatSqlActions(prefix, suffix).as[Int]
+      }
+
+      def dropAttributeTempTableAction() = {
+        sql"""drop table #${baseTableRow.tableName}_TEMP""".as[Int]
```

Comment:
this may be changed to "drop temporary table if exists ..." in gawb-766, which will be executed before and after we're done using this temporary table